portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -33,24 +34,29 @@
 /** Servlet that returns a programmable number of comments */
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
+  private final static int ALL_COMMENTS = -1;
+
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long maxComments = Long.parseLong(request.getParameter("maxComments"));
-
-    // a value of -1 indicates that ALL comments are wanted
-    if (maxComments == -1) maxComments = Long.MAX_VALUE;
+    String parameterMaxComments = getParameter(request, "maxComments", "");
+    int maxComments = 0;
+
+    try{
+      maxComments = Integer.parseInt(parameterMaxComments);
+    } catch (NumberFormatException ex) {
+      // If maxComments parameter is excluded or malformed, return all comments
+      maxComments = DataServlet.ALL_COMMENTS;
+    }
```

Comment:
What happens is a user updates the maximum comments but then reloads the page?

Reply:
I'm confused as to what you mean by 'updates maximum comments'? If you're asking about a user modifying the URL in the browser tab, and changing the value after '?maxComments=', then it'll reload with the number of comments specified.

Reply:
Just for the record as we discussed this over IM.

When a user changes the max comments drop down then reloads the page, it isn't persisted at the moment. We agreed to fix this in another change since the bug wasn't introduced in this one.